### PR TITLE
Don't add custom command to history if it starts with space

### PR DESCRIPTION
**Pull request by mark2185**
- **PR Description**
Sometimes you just want to run a one-off command and you don't want to have to look at it every time you run custom commands in the future.
Just like with shells, adding a space at the beginning doesn't append it to history.

- **Please check if the PR fulfills these requirements**

* [x] Cheatsheets are up-to-date (run `go run scripts/cheatsheet/main.go generate`)
* [x] Code has been formatted (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#code-formatting))
* [x] Tests have been added/updated (see [here](https://github.com/jesseduffield/lazygit/blob/master/pkg/integration/README.md) for the integration test guide)
* [x] Text is internationalised (see [here](https://github.com/jesseduffield/lazygit/blob/master/CONTRIBUTING.md#internationalisation))
* [x] Docs (specifically `docs/Config.md`) have been updated if necessary
* [x] You've read through your own file changes for silly mistakes etc

I'll come back to this and write the needed tests, but I'm opening so I don't forget, this way I can just procrastinate and postpone it til kingdom come :^)

### CODE REVIEW

File: pkg/gui/controllers/global_controller.go [hunk 1]
```diff
@@ -37,10 +39,12 @@ func (self *GlobalController) customCommand() error {
 		Title:               self.c.Tr.CustomCommand,
 		FindSuggestionsFunc: self.GetCustomCommandsHistorySuggestionsFunc(),
 		HandleConfirm: func(command string) error {
-			self.c.GetAppState().CustomCommandsHistory = utils.Limit(
-				lo.Uniq(append(self.c.GetAppState().CustomCommandsHistory, command)),
-				1000,
-			)
+			if !strings.HasPrefix(command, " ") {
```
Comment (jesseduffield) [inline pkg/gui/controllers/global_controller.go:42]: I'd extract out `!strings.HasPrefix(command, " ")` into a method called `shouldSaveCommand(command)` to make things clearer, and I'd add a comment explaining that this mimics existing behaviour in shells (I didn't realise this upon first reviewing).

File: pkg/integration/tests/custom_commands/basic_at_runtime.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
+package custom_commands
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/config"
+	. "github.com/jesseduffield/lazygit/pkg/integration/components"
+)
+
+var BasicAtRuntime = NewIntegrationTest(NewIntegrationTestArgs{
```
Comment (jesseduffield) [inline pkg/integration/tests/custom_commands/basic_at_runtime.go:8]: I would have two tests: one for the basic case of just adding a file, and one for the case of storing history. This test is combining the two and makes it a little harder to follow.

**Review by jesseduffield — CHANGES_REQUESTED** [review]
couple things :)

File: pkg/integration/tests/custom_commands/basic_at_runtime.go (hunk 2 above)
Comment (mark2185) [reply pkg/integration/tests/custom_commands/basic_at_runtime.go:8]: > one for the basic case of just adding a file, and one for the case of storing history

Hmmm just to confirm, one test for confirming the command _is_ saved, and another for confirming that it _isn't_ saved?

Comment (jesseduffield) [reply pkg/integration/tests/custom_commands/basic_at_runtime.go:8]: one test to confirm that the command is executed (e.g. creating a file) and a separate test to test the history logic (so you could rename the current test to be about history and make a separate smaller test that just asserts the command is executed)

**Comment by jesseduffield** [comment]
Awesome work :) I've just pushed a commit to keep the shouldSaveCommand function defined in the same file that it's called from, cos I doubt it will be needed elsewhere. I also squashed your tests commit with the commit that adds the code.

Once these tests pass I'll merge

**Pull request merged**